vault/core.go

```diff
@@ -236,6 +236,7 @@ type Core struct {
 	// about the seal we are migrating *from*.
 	migrationInfo *migrationInformation
 	sealMigrated  *uint32
+	migrationLock *int32
```

Comment:
I wonder if there is a reason why we always use `uint32` whenever we use this kind of locking.

Reply:
I switched to uber-atomic